### DISCUSSION TITLE
how to get the value like f(n+1)=g(f(n))

### DISCUSSION THREAD

**Discussion by xiaojbing** [idaholab/moose#23388] [Q&A General] [answered]
hello 
I want to ask, if I know the initial value of `f(0)`, and the fuction f(n+1)=g(f(n)), n is the time, g is a fuction. how to coupled f(n+1) to my Kernel?
thank you!!!

**Comment by GiudGiud** [comment, marked as answer]
Hi

Check out the doxygen for kernels, you can get the current time and the dt, then you can evaluate the function at the time you need
https://mooseframework.inl.gov/docs/doxygen/moose/classKernel.html

Guillaume

**Reply by xiaojbing** [reply]
It seems I need the old value f(n) if I want to evaluate f(n+1). I only know the initial value of f(n) and function. Specifically speaking,f(0) and g is known, `f(1)=g(f(0))`, then `f(2)=g(f(1))`, ...
I dont konw how to solve this.
thank you

**Reply by GiudGiud** [reply]
You can evaluate functions at any time you want.
Maybe rewrite your equation using MOOSE terminology.
What are f, g, n exactly?

**Reply by xiaojbing** [reply]
f is the value I want to couple to my kernel. g is a fuction about f. n is the time
